kairos-contracts/demo-contract/deposit-session/src/main.rs

```diff
@@ -0,0 +1,36 @@
+/*
+    Transfer native Casper tokens from the caller to the deposit smart contract.
+    Due to the purse access control in Casper 1.5.x, a temporary purse is created and funded
+    by the user first, to then be passed to the deposit contract.
+
+    Finally the temporary purse is emptied / all funds are transferred to the deposit contract's
+    purse.
+*/
+#![no_std]
+#![no_main]
+use casper_contract::contract_api::{account, runtime, system};
+use casper_types::{runtime_args, ContractHash, RuntimeArgs, URef, U512};
+
+#[no_mangle]
+pub extern "C" fn call() {
+    let contract_hash: ContractHash = runtime::get_named_arg("demo_contract");
+    let amount: U512 = runtime::get_named_arg("amount");
```

Comment:
I'm just wondering about the `constants` module which we are not using here: `RUNTIME_ARG_AMOUNT`

Reply:
I chose not to make the main contract a lib, it would be a bit odd if the test sessions depended on the main contract.